src/cpp/client.cpp

```diff
@@ -932,4 +884,78 @@ void Client::_unpack_dataset_metadata(DataSet& dataset,
                                       reply[i+1].str_len());
     }
     return;
+}
+
+TensorBase* Client::_get_tensorbase_obj(const std::string& name)
+{
+    std::string g_key = this->_build_tensor_key(name, true);
+
+    CommandReply reply = this->_redis_server->get_tensor(g_key);
+
+    std::vector<size_t> dims =
+        CommandReplyParser::get_tensor_dims(reply);
+
+    TensorType type =
+         CommandReplyParser::get_tensor_data_type(reply);
+
+    std::string_view blob =
+        CommandReplyParser::get_tensor_data_blob(reply);
+
+    if(dims.size()<=0)
+        throw std::runtime_error("The number of dimensions of the "\
+                                "fetched tensor are invalid: " +
+                                std::to_string(dims.size()));
+
+    for(size_t i=0; i<dims.size(); i++) {
+        if(dims[i]<=0) {
+        throw std::runtime_error("Dimension " +
+                                 std::to_string(dims[i]) +
+                                 "of the fetched tensor is "\
+                                 "not valid: " +
+                                 std::to_string(dims[i]));
+        }
+    }
+
+    TensorBase* ptr;
+    switch(type) {
+        case TensorType::dbl :
+            ptr = new Tensor<double>(g_key, (void*)blob.data(), dims,
+                                     type, MemoryLayout::contiguous);
+
+            break;
+        case TensorType::flt :
+            ptr = new Tensor<float>(g_key, (void*)blob.data(), dims,
+                                    type, MemoryLayout::contiguous);
+            break;
+        case TensorType::int64 :
+            ptr = new Tensor<int64_t>(g_key, (void*)blob.data(), dims,
+                                     type, MemoryLayout::contiguous);
+            break;
+        case TensorType::int32 :
+            ptr = new Tensor<int32_t>(g_key, (void*)blob.data(), dims,
+                                      type, MemoryLayout::contiguous);
+            break;
+        case TensorType::int16 :
+            ptr = new Tensor<int16_t>(g_key, (void*)blob.data(), dims,
+                                      type, MemoryLayout::contiguous);
+            break;
+        case TensorType::int8 :
+            ptr = new Tensor<int8_t>(g_key, (void*)blob.data(), dims,
+                                     type, MemoryLayout::contiguous);
+            break;
+        case TensorType::uint16 :
+            ptr = new Tensor<uint16_t>(g_key, (void*)blob.data(), dims,
+                                       type, MemoryLayout::contiguous);
+            break;
+        case TensorType::uint8 :
+            ptr = new Tensor<uint8_t>(g_key, (void*)blob.data(), dims,
+                                      type, MemoryLayout::contiguous);
+            break;
+        default :
+            throw std::runtime_error("The tensor could not be "\
+                                     "retrieved in "\
+                                     "client.get_tensorbase_obj().");
```

Comment:
Can we make this error message more informative? Something about the fact the type was was not recognized as a TensorType or similar?